### PR TITLE
Issue 679 - Add support for DWG importer

### LINKED ISSUE #679: DWG Object Handles imported via ODA do not match Navis
### Description

When importing DWGs via ODA, the Entity ID in the metadata is different to that shown in the Navis tree, as is the layer name in the Tree.

- This is happening in: both 
- This requires a hotfix to production: no 

### Steps to replicate
1. Import the file in the attached zip
[boxes.zip](https://github.com/3drepo/3drepobouncer/files/15009231/boxes.zip)

### Current Behaviour
- The Entity IDs are 253 & 261, whereas in Navisworks they are 30C and 308 (the same as in AutoCAD, where the file was originally authored).

Use the `LIST` command to check the Handles of selected objects in AutoCAD.

### Expected Behaviour
The Entity IDs should match those in Navisworks.

### Pictures/ Gifs of the error
image

This issue is part of: https://github.com/3drepo/3D-Repo-Product-Team/issues/387

### Solution

From the investigation below, the DGN type itself is not capable of holding the Autodesk metadata required to marry up the records between different Autodesk files.

The solution is to write a dedicated DWG importer, instead of relying on the DWG to DGN conversion.

### Tasks

Note this task list is incomplete - more tasks will be added as we import more complex files and compare to Navis.

- [x] Change `FileProcessorDwg` inheritance to remove reliance on `FileProcessorDgn`
- [x] Add a new `DataProcessorDwg` class to handle the import
- [x] Update cmake files to support loading DWGs natively into an `OdDbDatabase`
- [x] Ensure we can vectorize geometry from the `OdDbDatabase`
- [x] Ensure that material conversion methods are called
- ~~[ ] Initialise the palette for colour conversion~~
- [x] Find the base type of the `OdGiDrawable` that can be used to extract hierarchical information
- [x] Make sure colour by pen works
- [x] Make sure colour by layer works
- [x] Make sure colour by block works
- [x] Make default material settings look better (no specular highlights etc)
- [x] Set up layers to match hierarchy shown in Navisworks
- [x] Hook up metadata
- [x] Make sure blocks work
- [x] Make sure class names come through
- [x] Don't import geo position markers (or at least stop them appearing as part of another node)
- [x] Create unit tests

### PR DESCRIPTION
**Pull request by sebjf**
This fixes #679 - DWG object handles do not match Navis.

This PR adds a dedicated DWG importer using the Drawings SDK, replacing the previous importer which converted the DWG to a DGN, and then imported that using the DGN importer.

The new importer preserves the Handle Ids from AutoCAD, and better matches the Navisworks tree.

#### New classes

* The new importer follows the pattern of the existing ODA importers: a `FileProcessor` reads the file into an `OdDatabase`, which is then vectorised by a Graphics System View subclass (`DataProcessorDwg`), which sets up the collector layers and metadata, before the Gs tesselates the geometry into the `DataProcessor` superclass's methods.

#### View
* The importer always imports the ModelSpace view.

#### Blocks
* The importer creates tree nodes for all primitives outside of Blocks, and Block References. Block References take the name of their Block. All nested Block Entities are collapsed to the top level Block Reference.
* Where Block Entities are assigned to different layers, tree nodes are created for the top-level Block Reference under each relevant layer. This allows the user to select & toggle elements by Layer regardless of Block membership, mirroring the behaviour of AutoCAD and Navisworks.

#### Names
* Leaf nodes take the name of the type of primitive. Known class names are converted to a user-friendly name, mirroring the behaviour of Navisworks.

#### Metadata
* Entities have only one metadata entry (`Entity Handle::Value`), which contains the hex encoded handle of the object. This handle can be seen in AutoCAD with the `LIST` command.

#### Colours
* `ODCOLORREF` has been replaced with `OdCmEntityColor` in the `MaterialColours` struct across all ODA importers.
* This is because this struct is part of the `DataProcessor` superclass which is used in a number of places.
* The change was made because the `ODCOLORREF` macros are encoding dependent, and a different coding was turning up for DWGs leading to colour channels being swapped. `OdCmEntityColor` should handle this transparently.

#### CMake
* New dependencies have been added to `FindODA.cmake`
* Additionally the `ODA_LIB_NAMES` variable now has each entry on its own line as is done for the sources and headers, to make diff/change control easier

#### Unit Tests
* Two new unit tests for DWGs have been added, to check the tree logic, and to check that password protected DWGs fail with an appropriate error.

### Things to note

1. A new error code has been added for the password protected case. The code is 37. This code will also be used by #667.
2. The handle entities created by the new importer are "correct", however they do not match those created by the NWD importer or the (Navis) plugin. The plugins and NWD importer will need to be [changed](https://github.com/3drepo/3D-Repo-Product-Team/issues/387#issuecomment-2074977327) before cross referencing will work.

### CODE REVIEW

File: bouncer/src/repo/error_codes.h
Comment (sebjf) [reply bouncer/src/repo/error_codes.h:93]: Added! https://github.com/3drepo/3drepo.io/pull/4969

**Review by carmenfan — APPROVED** [review]
This is fine and good to merge

Just waiting on #685 so we can merge that in first (incase we need to do a hotfix!)

**Review by carmenfan — APPROVED** [review]

**Review by carmenfan — CHANGES_REQUESTED** [review]
@sebjf  travis is upset after merge from staging 😢

**Comment by sebjf** [comment]
@carmenfan fixed!

**Review by carmenfan — APPROVED** [review]

**Pull request merged**
